pkg/systemd/overview-cards/systemInformationCard.jsx

```diff
@@ -83,6 +83,18 @@ export class SystemInfomationCard extends React.Component {
                         self.setState({ hardwareText: vendor + " " + name });
 
                     self.setState({ assetTagText: fields.product_serial || fields.chassis_serial || undefined });
+
+                    // The Raspberry Pi 3 Model B Rev 1.2 has /sys/class/dmi/id/* but it's filled with "Unknown" values when using
+                    // U-boot with an EFI payload as is the case with Fedora Server / IoT.
+                    machine_info.devicetree_info()
+                            .then(dev_fields => {
+                                if (!fields.product_serial) {
+                                    self.setState({ assetTagText: dev_fields.serial });
+                                }
+                                if ((!vendor && !name && dev_fields.model) || (dev_fields.model && vendor && vendor.includes("Unknown"))) {
```

Comment:
Let's please rather filter this out in pkg/lib/machine-info.js. That already does a similar hack for the dreaded "To be filled by O.E.M." value. Then you can factorize this into some `is_invalid_value()` helper and use it in DMI and DT.

Reply:
Dropping this out because of this comment https://github.com/cockpit-project/cockpit/issues/16972#issuecomment-1041455088

As a nice exercise I've build my own u-boot with a patch from #u-boot to see if I can easily fix it in upstream.